parser-typechecker/src/Unison/Codebase/SqliteCodebase/Operations.hs

```diff
@@ -593,35 +597,79 @@ namesAtPath path = do
         Nothing -> (mempty, [(n, ref)])
         Just stripped -> ([(Name.makeRelative stripped, ref)], mempty)
 
-saveRootNamesIndex :: Names -> Transaction ()
-saveRootNamesIndex Names {Names.terms, Names.types} = do
-  let termNames :: [(S.NamedRef (C.Referent.Referent, Maybe C.Referent.ConstructorType))]
-      termNames = Rel.toList terms <&> \(name, ref) -> S.NamedRef {reversedSegments = nameSegments name, ref = splitReferent ref}
-  let typeNames :: [(S.NamedRef C.Reference.Reference)]
-      typeNames =
-        Rel.toList types
-          <&> ( \(name, ref) ->
-                  S.NamedRef {reversedSegments = nameSegments name, ref = Cv.reference1to2 ref}
-              )
-  Ops.rebuildNameIndex termNames typeNames
+-- | Update the root namespace names index which is used by the share server for serving api
+-- requests.
+--
+-- This version should be used if you've already got the root Branch pre-loaded, otherwise
+-- it's faster to use 'updateNameLookupIndexFromV2Branch'
+updateNameLookupIndexFromV1Branch :: Branch Transaction -> Sqlite.Transaction ()
```

Comment:
This version is not currently used, since we don't do anything with name indices on the client, but I want to keep it around for when we do.